docs/apache-airflow/howto/timetable.rst

```diff
@@ -0,0 +1,261 @@
+  like ``infer_data_interval``'s return value. This describes the next run's
+  data interval.
+* ``run_after: A ``pendulum.DateTime`` instance that tells the scheduler when
+  the DAG run can be scheduled.
+
+.. note::
+
+    In case you're wondering---yes, the argument and return value of
+    ``infer_data_interval`` are also internally combined into a ``DagRunInfo``.
+
+A ``DagRunInfo`` can be created like this:
+
+.. code-block:: python
+
+    info = DagRunInfo(
+        data_interval=DataInterval(start=start, end=end),
+        run_after=run_after,
+    )
+
+But since we typically wan to schedule a run as soon as the data interval ends,
+``end`` and ``run_after`` above are generally the same. ``DagRunInfo`` therefore
+provides a shortcut for this:
+
+.. code-block:: python
+
+    info = DagRunInfo.interval(start=start, end=end)
+    assert info.data_interval.end == info.run_after  # Always True.
+
+For reference, here's our DAG file in its entirety:
+
+.. exampleinclude:: /../../airflow/example_dags/example_workday_timetable.py
+    :language: python
+    :start-after: [START howto_timetable]
+    :end-before: [END howto_timetable]
+
+
+DAG Serialization and Parameterized Timetables
+----------------------------------------------
+
+Sometimes we need to pass some run-time arguments to the timetable. Continuing
+with out ``AfterWorkdayTimetable``, maybe we may have DAGs running on different
+timezones, and we want to schedule some DAGs at 8am the next day, instead of
+on midnight. Instead of creating a separate timetable for each purpose, we'd
+want to do something like:
+
+.. code-block:: python
+
+    from datetime import timedelta
+
+    from pendulum import DateTime, Time
+
+
+    class SometimeAfterWorkdayTimetable(Timetable):
+        def __init__(self, schedule_at: Time) -> None:
+            self._schedule_at = schedule_at
+
+        def next_dagrun_info(self, last_automated_dagrun, restriction):
+            ...
+            end = start + timedelta(days=1)
+            return DagRunInfo(
+                data_interval=DataInterval(start=start, end=end),
+                run_after=DateTime.combine(end.date(), self._schedule_at).replace(
+                    tzinfo=end.tzinfo
+                ),
+            )
+
+However, since the timetable is a part of the DAG, we need to tell Airflow how
+to serialize it with the context we provide in ``__init__``. This is done by
+implementing two additional methods on our timetable class:
+
+.. code-block:: python
+
+    def serialize(self) -> Dict[str, Any]:
+        return {"schedule_at": self._schedule_at.isoformat()}
+
+
+    @classmethod
+    def deserialize(cls, value: Dict[str, Any]) -> Timetable:
+        return cls(Time.fromisoformat(value["schedule_at"]))
+
+When the DAG is being serialized, ``serialize`` is called to obtain a
+JSON-serializable value. That value is passed to ``deserialize`` when the
+serialized DAG is accessed by the scheduler to reconstruct the timetable.
```

Comment:
(Looks fine, but this code isn't landed yet, right?)

Reply:
Yes, I’m anticipating the actual code change in #17414.

Reply:
There might be a thing we need to add there about "registering" the class too, but we can add that to the docs when we decide.